core/modules/entity/entity.module

```diff
@@ -739,15 +739,79 @@ function entity_view_mode_prepare($entity_type, $entities, $view_mode, $langcode
   return $entities_by_view_mode;
 }
 
+/**
+ * Implements hook_preprocess().
+ *
+ * Add template suggestions for an entity based on view modes if they do not
+ * already exist.
+ *
+ * The eventual order of the entity's theme suggestions are:
+ * - entity-type__bundle (lowest priority = first in the array)
+ * - entity-type__bundle__view-mode
+ * - entity-type__id
+ * - entity-type__id__view-mode (highest priority = last in the array)
+ */
+function entity_preprocess(&$variables, $hook) {
+  if (!empty($variables['elements']['#view_mode'])) {
+
+    // Sometimes a rendered entity is passed into another theme function, in
+    // which case we should not process. But also account that #theme may be a
+    // hook suggestion itself. For example. #theme = 'comment__node_type'
+    // and $hook = 'comment'.
+    if ($variables['elements']['#theme'] != $hook && strpos($variables['elements']['#theme'], $hook . '__') !== 0) {
+      return;
+    }
+
+    $view_mode = $variables['elements']['#view_mode'];
+    foreach ($variables['elements'] as $item) {
+      if (is_a($item, 'Entity')) {
+        $entity = $item;
```

Comment:
If my above suggestion doesn't work, then we could introduce a `break;` here to avoid looping through the rest of the stuff.